playbooks/adjust-cinder-backend.yml

```diff
@@ -0,0 +1,239 @@
+---
+
+# This playbook is used to meet the ursula 3.1.0 upgrade pre-conditions for
+# cinder service. In this playbook following configurations will be adjusted to
+# enable cinder for multi-backend support.
+#
+# 1: Some previous versions of ursula configured cinder service based on host it
+# was running on. Since we are using Ceph for cinder backend, cinder service must
+# be configured to unique host (i.e host=ceph) to make cinder services HA.
+#
+# 2: Adjust cinder backend name from rbd_volumes to new name based on the ceph backend type.
+#    - rbd_ssd
+#    - rbd_hybrid
+#
+# 3: Create a cinder volume_type based on backend name(set as extra_specs) and
+#    set it as a default. Also adjust existing volumes to this default volume_type.
+#    - CEPH_SSD
+#    - CEPH_HYBRID
+#
+
+- name: adjust cinder unique host and cinder backend for multi-backend support
+  hosts: cinder_volume
+  any_errors_fatal: true
+  max_fail_percentage: 1
+  gather_facts: false
+  environment: "{{ env_vars|default({}) }}"
+  vars:
+    date_time: "{{ lookup('pipe', 'date +%Y%m%d-%H%M%S') }}"
+    set_cinder_host: false
+    update_cinder_backend: false
+    set_default_volume_type: false
+    legacy_cinder_backend_name: rbd_volumes
+
+  roles:
+    - ../roles/endpoints
+
+  pre_tasks:
+    - fail: msg="all.yml must not contain cinder.common_volume_name attribute."
+      when: cinder.common_volume_name is defined
+      run_once: true
+
+    - fail: msg="Ceph is not enabled for this cluster."
+      when: not ceph.enabled|default('False')|bool
+      run_once: true
+
+    - fail: msg="Cinder is not enabled for this cluster."
+      when: not cinder.enabled|default('False')|bool
+      run_once: true
+
+    - name: dump cinder db
+      mysql_db:
+        name: cinder
+        state: dump
+        target: /backup/cinder-preupgrade.{{ date_time }}.sql
+      run_once: true
+      delegate_to: "{{ groups['db'][0] }}"
+      changed_when: false
+
+  handlers:
```

Comment:
Generally I like to define handlers after the play, as it keeps with the "top down" execution flow, but it doesn't technically matter.